packages/lodestar/test/unit/network/gossip/gossipsub.test.ts

```diff
@@ -1,66 +1,73 @@
 import {generateEmptySignedBlock} from "../../../utils/block";
 import {config} from "@chainsafe/lodestar-config/lib/presets/minimal";
-import {Message} from "libp2p-gossipsub/src/message";
+import {InMessage} from "libp2p-interfaces/src/pubsub";
 import {getGossipTopic} from "../../../../src/network/gossip/utils";
-import {GossipEvent, ExtendedValidatorResult} from "../../../../src/network/gossip/constants";
+import {ExtendedValidatorResult, GossipEvent} from "../../../../src/network/gossip/constants";
 import {IGossipMessageValidator} from "../../../../src/network/gossip/interface";
 import sinon from "sinon";
 import {LodestarGossipsub} from "../../../../src/network/gossip/gossipsub";
 import {WinstonLogger} from "@chainsafe/lodestar-utils";
-import {expect} from "chai";
-import {createPeerId} from "../../../../src/network";
-import {compress} from "snappyjs";
+import {expect, assert} from "chai";
+import {Libp2p} from "libp2p-gossipsub/src/interfaces";
+import {createNode} from "../../../utils/network";
+import {GossipEncoding} from "../../../../src/network/gossip/encoding";
+import {ERR_TOPIC_VALIDATOR_REJECT} from "libp2p-gossipsub/src/constants";
 
 const forkValue = Buffer.alloc(4);
 
 describe("gossipsub", function () {
   const sandbox = sinon.createSandbox();
   let validator: IGossipMessageValidator;
   let gossipSub: LodestarGossipsub;
-  let message: Message;
+  let message: InMessage;
 
   beforeEach(async function () {
     const signedBLock = generateEmptySignedBlock();
     message = {
-      data: compress(Buffer.from(config.types.SignedBeaconBlock.serialize(signedBLock))),
-      from: Buffer.from("0"),
-      seqno: Buffer.from("0"),
-      topicIDs: [getGossipTopic(GossipEvent.BLOCK, forkValue)],
+      data: config.types.SignedBeaconBlock.serialize(signedBLock),
+      from: "0",
+      receivedFrom: "0",
+      seqno: new Uint8Array(),
+      topicIDs: [getGossipTopic(GossipEvent.BLOCK, forkValue, GossipEncoding.SSZ)],
+      signature: undefined,
+      key: undefined,
     };
     validator = {} as IGossipMessageValidator;
-    const registrar = {
-      handle: (): null => null,
-      register: (): string => "",
-      unregister: (): boolean => false,
-    };
-    const peerId = await createPeerId();
-    gossipSub = new LodestarGossipsub(config, validator, new WinstonLogger(), peerId, registrar, {});
+    const multiaddr = "/ip4/127.0.0.1/tcp/0";
+    const libp2p = await createNode(multiaddr);
+    gossipSub = new LodestarGossipsub(config, validator, new WinstonLogger(), (libp2p as unknown) as Libp2p, {});
```

Comment:
Can you try to typecast without the `unknown`?

Reply:
Nvm, related to above comment

https://github.com/ChainSafe/lodestar/pull/1541#discussion_r499922451